pkg/asset/installconfig/aws/basedomain.go

```diff
@@ -22,28 +22,33 @@ func IsForbidden(err error) bool {
 
 // GetBaseDomain returns a base domain chosen from among the account's
 // public routes.
-func GetBaseDomain() (string, error) {
+func GetBaseDomain() (string, string, error) {
 	session, err := GetSession()
 	if err != nil {
-		return "", err
+		return "", "", err
 	}
 
 	logrus.Debugf("listing AWS hosted zones")
 	client := route53.New(session)
-	publicZoneMap := map[string]struct{}{}
-	exists := struct{}{}
+	publicZoneMap := map[string][]string{}
 	if err := client.ListHostedZonesPages(
 		&route53.ListHostedZonesInput{},
 		func(resp *route53.ListHostedZonesOutput, lastPage bool) (shouldContinue bool) {
 			for _, zone := range resp.HostedZones {
 				if zone.Config != nil && !aws.BoolValue(zone.Config.PrivateZone) {
-					publicZoneMap[strings.TrimSuffix(*zone.Name, ".")] = exists
+					zoneName := strings.TrimSuffix(aws.StringValue(zone.Name), ".")
+					zoneID := strings.TrimPrefix(aws.StringValue(zone.Id), "/hostedzone/")
+					if _, ok := publicZoneMap[zoneName]; ok {
```

Comment:
I dunno how consistent we are about this, but [you can append to a `nil` slice][1], so this block could probably be:

```go
publicZoneMap[zoneName] = append(publicZoneMap[zoneName], zoneID)
```

without the `if`/`else` logic (and CI will tell us if that doesn't work ;)

[1]: https://play.golang.org/p/UHZiXYh6tlL

Reply:
Tested this and I can confirm it works as expected!